numba/np/arrayobj.py

```diff
@@ -4958,12 +4958,93 @@ def np_flip(a):
         raise errors.TypingError("Cannot np.flip on %s type" % a)
 
     def impl(a):
-        sl = _build_slice_tuple(a.ndim)
+        sl = _build_flip_slice_tuple(a.ndim)
         return a[sl]
 
     return impl
 
 
+@intrinsic
+def _build_full_slice_tuple(tyctx, sz):
+    """Creates a sz-tuple of full slices"""
+    size = int(sz.literal_value)
+    tuple_type = types.UniTuple(dtype=types.slice2_type, count=size)
+    sig = tuple_type(sz)
+
+    def codegen(context, builder, signature, args):
+        def impl(length, empty_tuple):
+            out = empty_tuple
+            for i in range(length):
+                out = tuple_setitem(out, i, slice(None, None))
+            return out
+
+        inner_argtypes = [types.intp, tuple_type]
+        inner_sig = typing.signature(tuple_type, *inner_argtypes)
+        ll_idx_type = context.get_value_type(types.intp)
+        # Allocate an empty tuple
+        empty_tuple = context.get_constant_undef(tuple_type)
+        inner_args = [ll_idx_type(size), empty_tuple]
+
+        res = context.compile_internal(builder, impl, inner_sig, inner_args)
+        return res
+
+    return sig, codegen
+
+
+@overload(np.array_split)
+def np_array_split(ary, indices_or_sections, axis=0):
+    # If this statement is put at the top of the file, numba fails to import
+    from numba.typed import List
```

Comment:
I think that this should just use the `reflected list` even though its deprecated. At some point there'll be a global switch over in Numba's type inference from reflected to typed list, mixing what gets returned from numba functions now may cause confusion/is inconsistent.

Reply:
Would that just be switching `out = List.empty_list(...)` to `out = list()`?